ui/src/app/components/confirm-delete-modal.tsx

```diff
@@ -29,47 +19,60 @@ export interface ConfirmDeleteModalProps {
 
 export interface ConfirmDeleteModalState {
   loading: boolean;
+  deleteDisabled: boolean;
 }
 
 export class ConfirmDeleteModal
   extends React.Component<ConfirmDeleteModalProps, ConfirmDeleteModalState> {
 
   constructor(props: ConfirmDeleteModalProps) {
     super(props);
-    this.state = {loading: false};
+    this.state = {
+      loading: false,
+      deleteDisabled: true,
+    };
   }
 
   emitDelete(): void {
     this.setState({loading: true});
     this.props.receiveDelete();
   }
 
+  validateDeleteText = (event) => {
+    event.toLowerCase().match('delete') ?
+      this.setState({deleteDisabled: false}) :
+      this.setState({deleteDisabled: true});
+  }
+
   render() {
     return <Modal loading={this.state.loading}>
-        <ModalTitle style={{lineHeight: '28px'}}>
```

Comment:
This modal component is not just used for Workspaces, but for other child resource types as well, hence why it was parameterized by type here. For example, try creating a notebook or cohort and then deleting it. This should make the bug clear. The scope of the ticket only applies to workspaces, it should not affect the deletion behavior of other resources.